dev/MagentoTests/Integration/InventoryInStorePickupSalesAdminUi/NotifyPickupControllerTest.php

```diff
@@ -0,0 +1,134 @@
+            return;
+        }
+
+        /**
+         * Create a customer and login
+         */
+        $this->customerFixture = new CustomerFixture(CustomerBuilder::aCustomer()->withAddresses(
+            AddressBuilder::anAddress()->asDefaultBilling(),
+            AddressBuilder::anAddress()->asDefaultShipping()
+        )->build());
+        $this->customerFixture->login();
+
+        /**
+         * Create a product with 5 qty
+         */
+        $this->productFixture = new ProductFixture(
+            ProductBuilder::aSimpleProduct()
+                ->withPrice(10)
+                ->withStockQty(5)
+                ->withIsInStock(true)
+                ->build()
+        );
+
+        /**
+         * Order 5 qty
+         */
+        $checkout = CustomerCheckout::fromCart(
+            CartBuilder::forCurrentSession()
+                ->withSimpleProduct(
+                    $this->productFixture->getSku(),
+                    5
+                )
+                ->build()
+        );
+        $order = $checkout->placeOrder();
+        $extensionAttributes = $order->getExtensionAttributes();
+        $extensionAttributes->setPickupLocationCode('abc123');
+        $order->setExtensionAttributes($extensionAttributes);
+        $order->save();
+
+        $this->assertGreaterThan(0, strlen($order->getIncrementId()), 'the order does not have a valid increment_id');
+        $this->assertIsNumeric($order->getId(), 'the order does not have an entity_id');
+
+        /**
+         * Load the product fresh, and confirm it has had stock decremented (because reservations are disabled)
+         */
+        $product = Bootstrap::getObjectManager()
+            ->get(\Magento\Catalog\Model\ProductFactory::class)
+            ->create()
+            ->loadByAttribute('sku', $this->productFixture->getSku());
+        $this->assertEquals($product->getId(), $this->productFixture->getId(), 'The product failed to load');
+        $this->assertEquals(0, $product->getQty(), 'The product should have qty=0 after the order');
+        $this->assertEquals(0, $product->getIsInStock(), 'The product should have is_in_stock=0 after the order');
+
+        /**
+         * Dispatch the notfiyPickup controller
+         *
+         * It should respond that the customer has been notified
+         *
+         * The plugin will force every case to be a "true" result for pickup being available, the only thing that
+         * disqualifies a product from being able to be picked up in the original implementation is it being OOS
+         *
+         * @link https://github.com/magento/inventory/blob/develop/InventoryInStorePickupSalesAdminUi/Controller/Adminhtml/Order/NotifyPickup.php
+         */
+        $this->getRequest()->setParam(
+            'order_id',
+            $order->getId()
+        );
+        $this->dispatch('backend/sales/order/notifyPickup');
+        $this->assertSessionMessages(
```

Comment:
That should do it, I guess. We could potentially also assert the absence of the error message "The order is not ready for pickup", but that would probably be duplicate.

Reply:
https://github.com/AmpersandHQ/magento2-disable-stock-reservation/pull/109/commits/5c99009d6b097ef6a1686996029503dd022c8142

no wrapper for what you suggested, but this is logically equivalent. as we'll verify we have 1 message, and the next assert quantifies what it should be.